tests/app/main/test_frameworks.py

```diff
@@ -2280,6 +2274,150 @@ def test_email_not_sent_if_api_call_fails(
             assert not send_email.called
 
     @mock.patch('dmutils.s3.S3')
+    @mock.patch('app.main.views.frameworks.send_email')
+    def test_framework_agreement_returned_having_signed_contract_variation_redirects_to_framework_dashboard(
```

Comment:
Technically this can't happen - suppliers can only agree to a variation *after* returning their framework agreement (the contract variation page isn't shown if they haven't returned framework agreement yet).

But I don't think it hurts to keep this test as a weird just-in-case?